Fix Summary result for Kubernetes application

Hosshii opened the pull request:
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #3007

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
-->
```release-note
Fix wrong summary result for Kubernetes application in case the image version was updated
```

khanhtc1202 commented:
Nice catch and well tested 👍 
nits, this PR provides a user-facing change, should write to the block something like: Fix wrong summary result for ... in case ...
/lgtm

nghialv commented on pkg/app/piped/planner/kubernetes/kubernetes_test.go:
```diff
@@ -437,3 +437,60 @@ func TestDetermineVersion(t *testing.T) {
+			}
+		})
+	}
+
```
nit: Remove empty line

Hosshii commented:
Sorry, I fixed the wrong place, so I fixed it again.
And I changed user-facing change to `Fix wrong summary result for Kubernetes application in case the image version was updated`

nghialv commented:
Sure.
/approve
